internal/gps/vcs_repo.go

```diff
@@ -24,6 +24,40 @@ type ctxRepo interface {
 	//ping(context.Context) (bool, error)
 }
 
+func newCtxRepo(s vcs.Type, ustr, path string) (r ctxRepo, err error) {
+	r, err = getVCSRepo(s, ustr, path)
+	if err != nil {
+		// if vcs could not initialize the repo due to a local error
+		// then the local repo is in an incorrect state. Remove and
+		// treat it as a new not-yet-cloned repo.
+
+		// TODO(marwan-at-work): pass a logger here to warn/give progress of the above comment.
```

Comment:
Let's be a little more vague about the solution in the TODO - passing a logger is not the preferred design goal, per my other comments.